File: leetcode/linkedlist/linkedlist.py
class Node:
        def __init__(self, data):
                self.data = data
                self.next = None
                self.prev = None
        

class MyLinkedList:
        def __init__(self):
                self.head = None
        
        def setNeighbour(self, source, dest):
                source.next = dest
                dest.prev = source
        
        def getIndex(self, indx):
                pass

        def addAtTheEnd(self):
                pass

        def addAtStart(self):
                pass
        
        def addAtIndex(self):
                pass

        def deleteAtIndex(self):
                pass

        def printList(self):
                temp = self.head
                while (temp):
                        print(temp.data)
                        temp = temp.next

elem = MyLinkedList()
elem.head = Node([])
p1 = elem.head
p2 = Node([1])
p3 = Node([3])
p4 = Node([1, 2])
p5 = Node([1])
p6 = Node([1])
p7 = Node([1])
elem.setNeighbour(p1, p2)
elem.setNeighbour(p2, p3)
elem.setNeighbour(p3, p4)
elem.setNeighbour(p4, p5)
elem.setNeighbour(p5, p6)
elem.setNeighbour(p6, p7)
elem.printList()

File: Javascript/linkedlist/linked-list.js
class Node {
    constructor(elem) {
        this.elem = elem;
        this.next = null;
        this.prev = null;
    }
}

class LinkedList {
    constructor() {
        this.head = null;
    }

    addNeighbour(src, dest) {
        src.next = dest;
        dest.prev = src;
    }

    printList() {
        let temp = this.head;
        while(temp) {
            console.log(temp.elem);
            temp = temp.next;
        }
    }

    setNodeAtStart(startingnode) {
        startingnode.next = this.head;
        this.head = startingnode;
    }

    setNodeAtEnd(endingnode) {
        let temp = this.head;
        while(temp) {
            if (temp.next == null || temp.next == undefined) {
                temp.next = endingnode;
                break;
            } else {
                temp = temp.next;
            }
        }
    }

    deleteNode(node) {
        this.head = node.next;
        node.remove;
    }

    setNodeAtPos(pos, node) {
        let counter = 1;
        let temporary = this.head;
        while(counter < pos) {
            if (counter+1==pos) {
                let tempNext = temporary.next;
                let tempPrev = temporary.next.prev;
                console.log(tempNext.elem);
                console.log(tempPrev.elem);
                node.next = tempNext;
                node.prev = tempPrev;
                temporary.next = node;
                counter++;
            } else {
                temporary = temporary.next;
                counter++;
            }

            if (temporary.next == null || temporary.next == undefined) {
                break;
            }
        }
    }
}

let llist = new LinkedList();
llist.head = new Node(23);
let second  = new Node(6);
let third = new Node(15);
llist.addNeighbour(llist.head, second);
llist.addNeighbour(second, third);
let firstNode = new Node(50);
let finalNode = new Node(70);
llist.setNodeAtStart(firstNode);
llist.setNodeAtEnd(finalNode);
llist.printList();
console.log('//////////');
newNode = new Node(45);
llist.setNodeAtPos(3, newNode);
console.log('/////////');
llist.printList();